MdeModulePkg/Include/UniversalPayload/PciRootBridges.h

```diff
@@ -47,6 +47,8 @@ typedef struct {
 ///
 typedef struct {
   UINT32                                        Segment;               ///< Segment number.
+  UNIVERSAL_PAYLOAD_PCI_ROOT_BRIDGE_APERTURE    Ecam;                  ///< Ecam: for multi segment RB
+

```

Comment:
this structure is defined by USF->UPL spec which we would like to maintain backward compatibility (the best case is binary structure compatible):
https://universalscalablefirmware.github.io/documentation/2_universal_payload.html#pci-root-bridges

Could we create another HOB for such multi-segment platform usage so no impact to old platforms or single segment platforms?

For example:
Create UefiPayloadPkg\Include\Guid\UplPciSegmentInfoHob.h and GetPciSegmentInfo() just return with this HOB data.

extern EFI_GUID  gUplPciSegmentInfoHobGuid;
typedef struct {
  UINTN                        Count;
  PCI_SEGMENT_INFO  SegmentInfo[0];
} UPL_PCI_SEGMENT_INFO_HOB;

Reply:
This should not cause any issue with older implementation. Older implementation does not refer to ecam field. All this patch would do is IF "reg" exists it will just fill up the struct. Then if you have multi segment, then include PciSegmentInfoLibAcpiBoardInfo.c which will fill up structs properly. In case of single segment maybe ecam value can come from PCD. So what we could do is default value comes from PCD and if reg is found, then DT ecam is added.

Reply:
thanks for supporting backward compatibility.